File: LEVEL2/기능개발.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer = {};
        ArrayList<Integer> arr = new ArrayList<>();
        ArrayList<Integer> time = new ArrayList<>();
        
        for(int i = 0; i < progresses.length; i++) {
            arr.add(progresses[i]);
            time.add(speeds[i]);
        }
        
        ArrayList<Integer> res = new ArrayList<>();
        
        while(arr.size() > 0) {
            int cnt = 0;
            
            for(int i = 0; i < arr.size(); i++) {
                arr.set(i, arr.get(i) + time.get(i));
            }
            
            for(int i = 0; i < arr.size(); i++) {
                if(i == 0 && arr.get(i) >= 100) {
                    cnt++;
                    arr.remove(i);
                    time.remove(i);
                    i--;
                }
                
                else if(cnt > 0 && arr.get(i) < 100) {
                    break;
                }
                
                else if(cnt > 0 && arr.get(i) >= 100) {
                    cnt++;
                    arr.remove(i);
                    time.remove(i);
                    i--;
                }
            }
            
            if(cnt > 0) {
                res.add(cnt);
            }
        }
        
        answer = new int[res.size()];
        
        for(int i = 0; i < res.size(); i++) {
            answer[i] = res.get(i);
        }
        
        return answer;
    }
}

File: LEVEL2/기능개발.cpp
#include<string>
#include<vector>
#include<queue>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> vc;
    queue<int> que;
    
    for(int i = 0; i < progresses.size(); i++) {
        int d = (100 - progresses[i]) / speeds[i];
        
        if((progresses[i] + d * speeds[i]) != 100) {
            d++;
        }
        
        vc.push_back(d);
    }
    
    que.push(vc[0]);
    
    for(int i = 1; i < vc.size(); i++) {
        if(que.front() >= vc[i]) {
            que.push(vc[i]);
        }
        
        else {
            answer.push_back(que.size());
            
            while(!que.empty()) {
                que.pop();
            }
            
            que.push(vc[i]);
        }
    }
    
    if(!que.empty()) {
        answer.push_back(que.size());
    }
    
    return answer;
}